### PR TITLE
C# documentation fixes

### PR DESCRIPTION
**Pull request by jtattermusch**
This should fix the documentation issues mentioned in  https://github.com/grpc/grpc/issues/6151.

### CODE REVIEW

**Comment by jtattermusch** [comment]
CC @stanley-cheung  @sreecha

File: docs/index.md [hunk 1]
```diff
@@ -561,17 +561,21 @@ var hello_proto = grpc.load(PROTO_PATH).helloworld;
   </div>
   <div id="csharp_generate">
 
-- To generate the code on Windows, we use `protoc.exe` from the `Google.Protobuf` NuGet package and `grpc_csharp_plugin.exe` from the `Grpc.Tools` NuGet package (both under the `tools` directory).
-Normally you would need to add the `Grpc.Tools` package to the solution yourself, but in this tutorial it has been already done for you. The following command should be run from the `examples/csharp/helloworld` directory: 
+Ready-to-use precompiled versions of `protoc` and `grpc_csharp_plugin` binaries for Windows, Linux and Mac OS X are available in the `Grpc.Tools` NuGet package (there are platform-specific subdirectories both under the `tools` directory).
+Normally you would need to add the `Grpc.Tools` package to the solution yourself, but in this tutorial it has been already done for you.
+
+The generate the code, the following command should be run from the `examples/csharp/helloworld` directory:
+
```
Comment (LisaFC) [inline docs/index.md]: To generate the code, run the following command from the `examples/csharp/helloworld` directory:

File: docs/index.md [hunk 2]
```diff
@@ -561,17 +561,21 @@ var hello_proto = grpc.load(PROTO_PATH).helloworld;
   </div>
   <div id="csharp_generate">
 
-- To generate the code on Windows, we use `protoc.exe` from the `Google.Protobuf` NuGet package and `grpc_csharp_plugin.exe` from the `Grpc.Tools` NuGet package (both under the `tools` directory).
-Normally you would need to add the `Grpc.Tools` package to the solution yourself, but in this tutorial it has been already done for you. The following command should be run from the `examples/csharp/helloworld` directory: 
+Ready-to-use precompiled versions of `protoc` and `grpc_csharp_plugin` binaries for Windows, Linux and Mac OS X are available in the `Grpc.Tools` NuGet package (there are platform-specific subdirectories both under the `tools` directory).
```
Comment (LisaFC) [inline docs/index.md]: ...subdirectories under the `tools` directory

File: docs/tutorials/basic/csharp.md [hunk 3]
```diff
@@ -35,13 +35,8 @@ $ git clone https://github.com/grpc/grpc.git
...
 All the files for this tutorial are in the directory  `examples/csharp/route_guide`.
-Open the solution `examples/csharp/route_guide/RouteGuide.sln` from Visual Studio (or Monodevelop on Linux).
-
-On Windows, you should not need to do anything besides opening the solution. All the needed dependencies will be restored
-for you automatically by the `Grpc` NuGet package upon building the solution.
-
-On Linux or Mac OS X, you will first need to install protobuf and gRPC C# native dependencies in order to be
-able to generate the server and client interface code and run the examples. Please follow the [How to use instructions](https://github.com/grpc/grpc/tree/{{ site.data.config.grpc_release_branch }}/src/csharp#how-to-use).
+Open the solution `examples/csharp/route_guide/RouteGuide.sln` from Visual Studio, Monodevelop or Xamarin Studio.
+For additional details, inspect the [How to use instructions](https://github.com/grpc/grpc/tree/{{ site.data.config.grpc_release_branch }}/src/csharp#how-to-use).
```
Comment (LisaFC) [inline docs/tutorials/basic/csharp.md]: For additional installation details, see the [How to use...

**Comment by jtattermusch** [comment]
Thanks for the comments, I think I've addressed them all.

File: docs/index.md [hunk 4]
```diff
@@ -561,17 +561,21 @@ var hello_proto = grpc.load(PROTO_PATH).helloworld;
   </div>
   <div id="csharp_generate">
 
-- To generate the code on Windows, we use `protoc.exe` from the `Google.Protobuf` NuGet package and `grpc_csharp_plugin.exe` from the `Grpc.Tools` NuGet package (both under the `tools` directory).
-Normally you would need to add the `Grpc.Tools` package to the solution yourself, but in this tutorial it has been already done for you. The following command should be run from the `examples/csharp/helloworld` directory: 
+Ready-to-use precompiled versions of `protoc` and `grpc_csharp_plugin` binaries for Windows, Linux and Mac OS X are available in the `Grpc.Tools` NuGet package (there are platform-specific subdirectories under the `tools` directory).
+Normally you would need to add the `Grpc.Tools` package to the solution yourself, but in this tutorial it has been already done for you.
+
+The generate the code, run the following command from the `examples/csharp/helloworld` directory:
```
Comment (LisaFC) [inline docs/index.md]: To generate the code...

File: docs/tutorials/basic/csharp.md [hunk 5]
```diff
@@ -108,24 +103,22 @@ message Point {
...
-
-- For Windows, the `Grpc.Tools` and `Google.Protobuf` NuGet packages contain the binaries you will need to generate the code.
-- For Linux or OS X, make sure you've followed the [How to use instructions](https://github.com/grpc/grpc/tree/{{ site.data.config.grpc_release_branch }}/src/csharp#how-to-use).
+If you want to run this yourself, the `Grpc.Tools` NuGet package contains the binaries you will need to generate the code.
 
 Once that's done, you can generate the C# code:
 
-- To generate the code on Windows, we use `protoc.exe` from the `Google.Protobuf` NuGet package and `grpc_csharp_plugin.exe` from the `Grpc.Tools` NuGet package (both under the `tools` directory).
-Normally you would need to add the `Grpc.Tools` package to the solution yourself, but in this tutorial it has already been done for you. Following command should be run from the `examples/csharp/route_guide` directory: 
+The generate the code, the following command should be run from the `examples/csharp/route_guide` directory:
```
Comment (LisaFC) [inline docs/tutorials/basic/csharp.md]: To generate the code...

**Comment by LisaFC** [comment]
Couple of nits, otherwise I think we're good to go. I'll merge it once they're fixed.

File: docs/index.md (hunk 4 above)
Comment (jtattermusch) [inline docs/index.md]: Done.

File: docs/tutorials/basic/csharp.md (hunk 5 above)
Comment (jtattermusch) [inline docs/tutorials/basic/csharp.md]: Done.

**Pull request merged**